src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
```

Comment:
Заміна let на const + прибираємо використання циклу forEach
```suggestion
  const sum = children.reduce((acc, curr) => {
    const mother = people.find(m => m.name === curr.mother);

    return acc + (curr.born - mother.born);
  }, 0)
```